scipy/optimize/_shuffled_complex_evolution.py

```diff
@@ -0,0 +1,1083 @@
+        the parameters in the form of an iterable.
+        ``args`` and ``kwargs`` are passed to the function via the usual
+        unpacking operators.
+    x0 : array_like
+        Parameter values with `mask==1` used in initial complex if
+        `iniflg==1`.
+    lb : array_like, sequence or `Bounds`
+        Lower bounds of parameters if ``ub`` given.
+        If `ub` is not given, then `lb` are the bounds of the parameters,
+        either 1) as an instance of the `Bounds` class, or 2) as ``(min, max)``
+        pairs for each element in ``x``, defining the finite lower and upper
+        bounds for the parameters of `func`.
+    ub : array_like, optional
+        Upper bounds of parameters.
+    mask : array_like, optional
+        Include (1, True) or exclude (0, False) parameters in minimization
+        (default: include all parameters). The number of parameters ``nopt`` is
+        ``sum(mask)``.
+    args : tuple, optional
+        Extra arguments passed to the function *func*. Note that ``args`` must
+        be iterable. `args=int`and `args=(int)` are not valid (with int being
+        any scalar variable) but should be `args=(int,)`.
+    kwargs : dict, optional
+        Extra keyword arguments passed to the function `func`.
+    sampling : string or array_like of strings, optional
+        Options for sampling random numbers. Options can be on of:
+
+            - 'half-open': same as 'right-half-open' [lb, ub)
+            - 'left-half-open': sample random floats in half-open
+              interval (lb, ub]
+            - 'right-half-open': sample random floats in half-open
+              interval [lb, ub)
+            - 'open': sample random floats in open interval (lb, ub)
+            - 'log': sample half-open interval [log(lb), log(ub)), which
+              samples better intervals spanning orders or magnitude such as
+              `lb=1e-9` and `ub=1e-4`
+
+        The default is 'half-open'.
+    maxn : int, optional
+        Maximum number of function evaluations allowed during minimization
+        (without polishing) (default: 1000).
+    kstop : int, optional
+        Maximum number of evolution loops before convergence (default: 10).
+    pcento : float, optional
+        Percentage change allowed in kstop loops before convergence
+        (default: 0.0001).
+    peps : float, optional
+        Value of normalised geometric range needed for convergence
+        (default: 0.001).
+    ngs : int, optional
+        Number of complexes (default: 2).
+    npg : int, optional
+        Number of points in each complex (default: `2*nopt+1`).
+    nps : int, optional
+        Number of points in each sub-complex (default: `nopt+1`).
+    mings : int, optional
+        Minimum number of complexes required if the number of complexes is
+        allowed to reduce as the optimization proceeds (default: `ngs`).
+    nspl : int, optional
+        Number of evolution steps allowed for each complex before complex
```

Comment:
If possible, we should try to rename arguments to something simple to understand (for example `n_complexey` instead of `ngs`). I have been guilty of bad argument names myself in the past but let's try to do better in future.